src/main/java/org/reaktivity/nukleus/amqp/internal/stream/AmqpServerFactory.java

```diff
@@ -2269,10 +2822,34 @@ private void onApplicationBegin(
                         AmqpReceiverSettleMode amqpReceiverSettleMode =
                             amqpReceiverSettleMode(amqpBeginEx.receiverSettleMode().get());
                         final String address = amqpBeginEx.address().asString();
+                        String otherAddress = null;
+                        switch (amqpRole)
+                        {
+                        case SENDER:
+                            otherAddress = addressTo;
+                            break;
+                        case RECEIVER:
+                            otherAddress = addressFrom;
+                            break;
+                        }
 
                         deliveryCount = initialDeliveryCount;
                         doEncodeAttach(traceId, authorization, name, channelId, handle, amqpRole, amqpSenderSettleMode,
-                            amqpReceiverSettleMode, address);
+                            amqpReceiverSettleMode, address, otherAddress);
+
+                        if (amqpRole == RECEIVER)
+                        {
+                            this.linkCredit = (int) (Math.min(bufferPool.slotCapacity(), initialBudget) /
+                                Math.min(bufferPool.slotCapacity(), defaultMaxFrameSize));
+                            minimum.value = Integer.MAX_VALUE;
+                            links.values().forEach(l -> minimum.value = l.linkCredit == 0 ? minimum.value :
+                                Math.min(l.linkCredit, minimum.value));
+                            incomingWindow = minimum.value == Integer.MAX_VALUE ? 0 : minimum.value;
```

Comment:
I think this part is going to change in your upcoming PR, so will not comment any further for now.